### ISSUE TITLE
"Type in the answer" gripes

### ISSUE THREAD

**Issue by EmilRehnberg** [ankidroid/Anki-Android#3912]
This is in regards to #3730 by @ospalh 

"Type in the answer" cards got an upgrade and I actually think the old version worked a little better.

1. "show answer" button does not show up while typing, the user has to tap the card browser to access the "show answer" button. This definitely needs to fixed IMO.

2. while the old version looked esoteric at first, having the type box at the bottom above the "show answer"-button, worked real good and it was fast (as touching and typing was at the bottom). Then again, I am not sure reverting now is the way to go (as I agree with changes made in principle).
Maybe I can fix the input box at the very bottom using HTML? `footer`-tag? or is there a nice CSS trick to fix it?

**Comment by ospalh** [comment]
The main motivation (“principle”, if you like) was to bring it more in line with desktop Anki. Also, with the change you can do pseudo-clozes. For example In my geography deck i have the input box integrated into the Question. “Ecuador has <input box> million inhabitants” and you are expected to type the number into the sentence. (Real clozes could, maybe, be made to work like this, but aren’t on Anki desktop.) 

Re 1.: You can use the keyboard’s return button. The answer button is hidden on purpose to save screen area. When you have a small phone and use it in landscape mode, you don’t see very much of the card with the keyboard visible, and would get even less with the answer button there, too.

Re 2.: Hmm. The input tag gets the id [typeans](https://github.com/ankidroid/Anki-Android/blob/develop/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java#L740). You could use that to manipulate it. You should probably use `input#typeans` rather than just `#typeans`, as that would change the comparison shown on the answer side, too.

**Comment by Musashi1986** [comment]
I also think that the old version worked better. I use a tablet with a keyboard attached to it and I have the following issues.

1. I cannot type characters with accents like á or ä unless I have a key with that specific characters (like ä, ö and ü in a german keyboard).

2. Before, when I got a card with a "type in the answer", I didn´t need to select the box before starting to write. Also, after checking the answer by pressing enter, before I could just press enter if I had it correct to go to the next one, while now I need to touch the screen in all cases.

**Comment by Egmorn** [comment]
Hello.

I can only support previous comment on regression for the new version of type box. I can understand the reason for the change but this change brought in major regression:

1) Answer box disappear while keyboard is appearing. This require one more click to make the keyboard disappear before validating answer (or pressing enter key on keyboard).

2) I use Ankidroid to learn Kanji so the "Drawing" tool is critical for learning. With previous version, input box appeared out of the drawing zone, allowing to key in the answer without deactivating the drawing tool. Now, the input box appear below the drawing tool, making it impossible to select without deactivation of the drawing tool (That is 2 additional click for deactivation and reactivation).

3) The selection of the box on my Samsung Galaxy Note 3 is very imprecise. I need to click multiple time on the bottom of the box to get the keyboard to appear. It gets even harder when using the stylus. I am getting the nick of it, but clearly there is an issue with the focus zone of the box.

**Comment by ospalh** [comment]
> 1) Answer box disappear while keyboard is appearing. This require one more click to make the keyboard disappear before validating answer (or pressing enter key on keyboard).

How is this one more click? Tapping the enter button on the software keyboard is one tap, as is tapping the the answer button w/o the keyboard

**Comment by Egmorn** [comment]
Except I am using Japanese keyboard, need to type twice (one to validate the word you type, and one more time as enter).
You should not assume the enter key replace the answer button as their are many different keyboard with many different settings.

**Comment by EmilRehnberg** [comment]
Thank you for replying so quickly @ospalh . 
I did not realize you could press Enter to go forward. This works for me. Thank you.
Also thank you very much for the id!

I had no idea about the "Whiteboard" feature @Egmorn. Thanks for chipping in!
I've been using google's hand-writing keyboard for drawing Kanji. But the built in whiteboard looks more useful.

Regarding @Egmorn 3. I do not have this problem on a Z3 compact or Nexus 7 (2013). So your issue could be a specific issue for more precise styluses.

@Musashi1986 regarding 2. I have always had to select a box when the new card shows, regardless of version. If you had a setting for this, please let me know because that sounds awesome.

**Comment by Musashi1986** [comment]
@EmilRehnberg I don´t know if I had a setting for it or not (if I chose something, it was several months ago and I forgot). I just know that for the last 3 months there were days when I could review all my cards for the day (half of them "type in the answer") without using the touch screen apart from choosing the deck; and it is no longer possible now.

**Comment by Egmorn** [comment]
For my focus issue, I have the issue with hand and with stylus. 
With hand it work more or less, but it feels like the focus zone is slightly offset down (If I click on the middle of the box it does not recognize the box selection 100%, but it work better if I aim for the bottom).
With Stylus, the issue seems to be the same, but the focus box seems to be even more reduced with a down and left offset. (I basically need to click on the bottom line and slightly on the left part to be sure to get the selection).

But that is only a 2 days use impression, I will confirm/infirm all that as I will use more this new update.

If someone else wants to test: Phone is a non rooted GN3 (SM-N9005) with Android 5.0

**Comment by eginhard** [comment]
@ospalh Is there any way to give the input box automatic focus and pop up the keyboard?

**Comment by ospalh** [comment]
@eginhard: I am against automatically showing the keyboard. That covers too much off the card content.

**Comment by eginhard** [comment]
Sorry, I had thought that was the case from reading the other messages, but
I noticed it wasn't shown before either - it only received focus without
showing the keyboard. Maybe the field's default size could be increased
then to make selecting it easier?

**Comment by Musashi1986** [comment]
@eginhard I think that is one of my issues. I'm not sure about the keyboard since I always use it in my tablet with a physical keyboard attached to it (so the keyboard does not pop up in the screen), but I liked how the focus was already in the input box, so that I could start rightaway typing the answer without having to select it. Maybe it could be included as an option in the configuration menu, so that we can decide whether we want the focus on the box or not.

**Comment by hssm** [comment]
A possible solution may be to [detect if a hardware keyboard is present](http://stackoverflow.com/questions/6654177/is-there-a-way-in-android-to-tell-if-a-users-device-has-an-actual-keyboard-or-no) and, if so, give the input field focus when showing a card.

**Comment by timrae** [comment]
Having to press the input box multiple times to get it to work is not so good, IIRC a user complained about this during the beta testing as well and they ended up setting some CSS to change the size. Can we make the default size bigger on AnkiDroid?

**Comment by maneshmistry** [comment]
I'd like to second the comment by @Egmorn on the whiteboard overlapping issue. It was very handy in the previous version to be able to select the answer field without needing to deactivate the whiteboard. There are many more clicks required now, whereas previously I could basically just leave the whiteboard on and power on through reviews.

Is there any way of having the answer field still selectable, even when whiteboard is active?

**Comment by timrae** [comment]
I think automatically reverting to the old behavior when whiteboard is enabled would be a good idea

**Comment by EmilRehnberg** [comment]
I agree wholeheartedly with @timrae 

I added some CSS to my set-up to cope with the recent update. 
```{css}
input#typeans {
  position: fixed;
  bottom: 0;
  left: 0;
  height: 2em;
}
```
1. put the text input-box on the bottom (if it does not work try to put the `input` aka `{{type:__}}` at the bottom of the mark-up)
2. increase the size of the input-box (by setting `height`, instead of the more sensible `font-size`. The latter does only work on Desktop)

**Comment by Musashi1986** [comment]
@EmilRehnberg  I decided to uninstall the new version and install the old one (2.4.4) since I like it more. I checked all my settings and I didn´t see anything special, so I don´t know why the input-box is always selected. I use an ASUS tablet+keyboard with android 4.4.2

**Comment by Egmorn** [comment]
Just checked the focus issue. After multiple test, there is clearly an offset on my Samsunge Note 3. The focus box is situated below the actual input box. So to get the keyboard, I need to click below the box. With finger it gets a bit easier cause the click area is large. With the stylus I clearly need to click Below the box (there is just a few pixel overlaping on the bottom)

**Comment by timrae** [comment]
I just released v2.5.2beta1, where the new input box is disabled by default. You can re-enable it if you like under advanced settings.

[Join the beta channel on Google play](https://play.google.com/apps/testing/com.ichi2.anki) to get automatic updates of beta versions, or you can install from the releases section of github.

**Comment by Egmorn** [comment]
Thanks for the quick solution.

**Comment by Musashi1986** [comment]
My experience with the new version.

New option disabled: I need to select the input box by touching it, I can write special characters like á or ä normally and after pressing enter to see the answer, I can press enter again to go to the next card (if I got it right, for any other option I need to touch the screen)

New option enabled: I need to select the input box by touching it (I can also press the tab key several times, but this is too much of a bother to do it everytime), I cannot write special characters like á or ä (I get something like ´a and ¨a, unless I have a key for that specific character) and I always need to touch the screen to go to the next card.

I would be very happy if it was possible to have the input box ready to be written on it when the new card comes out without having to touch it, as it somehow happens when I use the version 2.4.4

**Comment by ospalh** [comment]
> @Musashi1986: New option disabled: I need to select the input box by touching it (unlike 2.4.4)

Hmm. That’s odd. We tell that input box that it can get the focus, 2.5.2 the same as with 2.4.4. I guess i have to look into this a bit more.

**Comment by timrae** [comment]
@Musashi1986 is this with full-screen mode enabled by any chance?

**Comment by maneshmistry** [comment]
Currently testing 2.5.2beta1 on a Nexus 5 phone. Using the old input box as default works great for me here. Thank you very much for such a quick solution!

**Comment by Musashi1986** [comment]
@timrae No. I never enabled the full-screen mode.

**Comment by timrae** [comment]
@ospalh 

In 2.5.2 is it OK to omit the "disable typing answer" option?

**Comment by ospalh** [comment]
IDK.
You should get only one of the “disable typing” or “use input box” options. On old Androids, removing the former would mean there would always be the input box when there is a {{type:NN}}.
Also, the reason why i left the old code in there in the first place was issues with focusing input boxes. An i think that is related to the “input workaround” setting nearby.
